.github/workflows/vulnerability-check.yaml

```diff
@@ -0,0 +1,54 @@
+name: Vulnerability check
+on:
+  workflow_call:
+
+jobs:
+  trivy:
+    name: Trivy scan
+    runs-on: ubuntu-24.04
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Setup Go
+        uses: actions/setup-go@f111f3307d8850f501ac008e886eec1fd1932a34 # v5.3.0
+        with:
+          go-version: "~1.24"
+          check-latest: true
+
+      - name: Run make generate
+        run: make generate
+
+      - name: Install Trivy
+        run: curl -sfL https://raw.githubusercontent.com/aquasecurity/trivy/main/contrib/install.sh | sudo sh -s -- -b /usr/local/bin v0.61.0
+
+      - name: Run Trivy repo scan
+        run: trivy repository . --scanners "vuln,misconfig,secret" --ignorefile ./.trivyignore.yaml --format json --output trivy-report.json
+
+      - name: Generate GitHub summary
+        run: |
+          echo "# Trivy Scan Results" >> $GITHUB_STEP_SUMMARY
+          echo "" >> $GITHUB_STEP_SUMMARY
+          
+          HIGH_COUNT=$(grep -o '"Severity": "HIGH"' trivy-report.json | wc -l)
+          CRITICAL_COUNT=$(grep -o '"Severity": "CRITICAL"' trivy-report.json | wc -l)
+          MEDIUM_COUNT=$(grep -o '"Severity": "MEDIUM"' trivy-report.json | wc -l)
+          LOW_COUNT=$(grep -o '"Severity": "LOW"' trivy-report.json | wc -l)
+          
+          echo "## 📊 Summary" >> $GITHUB_STEP_SUMMARY
```

Comment:
Lovely :rocket: , here is the rendered version for the current state:
https://github.com/axoflow/axoflow-otel-collector-releases/actions/runs/14302704989/attempts/1#summary-40080031873